### PR TITLE
Extending the Functionality

### PR DESCRIPTION
**Pull request by kgrandly**
Add private and collaborative Spotify playlists and other features.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,15 +5,25 @@ Tool for Migration from Spotify to Yandex Music
 1) Install requirements: `pip3 install -r requirements.txt`
 2) Provide environment variables ([see configuration](#configuration))
 3) Run application: `python3 ./src/main.py`
+4) Log in to the open page in browser
```
Comment (ythosa) [inline README.md:8]: Let these steps (4,5,6) be included in the configuration, i.e in step 2

File: src/main.py [hunk 2]
```diff
@@ -25,15 +28,42 @@ def main():
 
         for spotify_track in spotify_playlist.tracks:
             artists = ' '.join(map(lambda artist: artist.name, spotify_track.artists))
-            track_name = f"{spotify_track.title} {artists}"
+            track_name = f'{spotify_track.title} {artists}'
 
+            attemp_count = 1
+            while attemp_count <= 3:
```
Comment (ythosa) [inline src/main.py]: Let's replace this `while` with a `for` loop and make constant `3` a constant with name like `ADD_TRACK_ATTEMPTS_NUMBER`

File: src/main.py [hunk 3]
```diff
@@ -25,15 +28,42 @@ def main():
...
+                try:
+                    yandex_track = yam.search_track(track_name)
+                    yandex_playlist = yam.add_to_playlist(yandex_playlist, yandex_track)
+                    break
+                except TypeError:
+                    print(f'Can\'t found track: "{track_name}" to add to playlist "{spotify_playlist.name}"')
+                    break
+                except Exception as e:
+                    print(f'Something went wrong... Attemp {attemp_count} {e}')
```
Comment (ythosa) [inline src/main.py]: Typo in `Attemp`

File: src/main.py [hunk 4]
```diff
@@ -25,15 +28,42 @@ def main():
...
+                yandex_track = yam.search_track(track_name)
+                yandex_playlist = yam.like_track(yandex_track)
+                break
             except TypeError:
-                print(f"Can't found track: '{track_name}' to add to playlist '{spotify_playlist.name}'")
+                print(f'Can\'t found track: "{track_name}" to add to playlist "{spotify_playlist.name}"')
+                break
             except Exception as e:
-                print(f"Something went wrong... {e}")
+                print(f'Something went wrong... Attemp {attemp_count} {e}')
```
Comment (ythosa) [inline src/main.py]: Type in `Attemp`

File: src/main.py [hunk 5]
```diff
@@ -25,15 +28,42 @@ def main():
...
+                    break
+                except Exception as e:
+                    print(f'Something went wrong... Attemp {attemp_count} {e}')
+                
+                attemp_count += 1
+    
+    spotify_playlist = spotify.get_saved_tracks()
+    print(f'Adding {spotify_playlist.name}')
+    
+    for spotify_track in spotify_playlist.tracks:
```
Comment (ythosa) [inline src/main.py:50]: Code duplication. 
You can make a function that will take the  `spotify.Playlist` and function which returns new playlist, i.e.: 

```python
def add_playlist(spotify_playlist, yandex_playlist, add_track):
   for spotify_track in spotify_playlist.tracks:
            artists = ' '.join(map(lambda artist: artist.name, spotify_track.artists))
            track_name = f'{spotify_track.title} {artists}'

            for attempt in range(ATTEMPTS_NUMBER):
                try:
                    yandex_track = yam.search_track(track_name)
                    yandex_playlist = add_track(yandex_playlist, yandex_track)
                    break
                except TypeError:
                    print(f'Can\'t found track: "{track_name}" to add to playlist "{spotify_playlist.name}"')
                    break
                except Exception as e:
                    print(f'Something went wrong... Attemp {attempt} {e}')
``` 

So for playlists will be: 

```python
add_playlist(spotify_playlist, yandex_playlist, lambda yandex_playlist, track: yam.add_to_playlist(yandex_playlist, track)
```

For favorites:
```python
add_playlist(spotify_playlist, yandex_playlist, lambda yandex_playlist, track: yam.like_track(track)
```

**Review by ythosa — COMMENTED** [review]
Thanks for the help! You're amazing :heart: 
But I need to fix something...

File: src/main.py (hunk 2 above)
Comment (kgrandly) [reply src/main.py]: Good idea

File: src/main.py (hunk 5 above)
Comment (kgrandly) [reply src/main.py:50]: My knowledge of Python is very poor, so you can do better!

File: src/main.py (hunk 2 above)
Comment (kgrandly) [reply src/main.py]: I noticed that 3 attempts are enough to resolve any error

File: README.md (hunk 1 above)
Comment (kgrandly) [reply README.md:8]: Maybe... but these steps come after the application is launched. You have to log in to Spotify and paste the redirect url into the terminal window.

**Pull request merged**